fe/fe-core/src/main/java/com/starrocks/scheduler/mv/IMTCreator.java

```diff
@@ -0,0 +1,231 @@
+                                                   long mvId, long dbId)
+            throws DdlException {
+        ExecPlan plan = Preconditions.checkNotNull(stmt.getMaintenancePlan());
+        OptExpression physicalPlan = plan.getPhysicalPlan();
+        MVOperatorProperty property =
+                Preconditions.checkNotNull(physicalPlan.getMvOperatorProperty(), "incremental mv must have property");
+        KeyInference.KeyPropertySet keys = property.getKeySet();
+        if (keys.empty()) {
+            throw UnsupportedException.unsupportedException("mv could not infer keys");
+        }
+        keys.sortKeys();
+        KeyInference.KeyProperty key = keys.getKeys().get(0);
+
+        // Columns
+        List<Column> columns = stmt.getMvColumnItems();
+
+        // Duplicate/Primary Key
+        Map<Integer, String> columnNames = plan.getOutputColumns().stream().collect(
+                Collectors.toMap(ColumnRefOperator::getId, ColumnRefOperator::getName));
+        Set<String> keyColumns = key.columns.getStream().mapToObj(columnNames::get).collect(Collectors.toSet());
+        for (Column col : columns) {
+            col.setIsKey(keyColumns.contains(col.getName()));
+        }
+        if (!property.getModify().isInsertOnly()) {
+            stmt.setKeysType(KeysType.PRIMARY_KEYS);
+        }
+
+        // Distribute Key, already set in MVAnalyzer
+        DistributionDesc distributionDesc = stmt.getDistributionDesc();
+        Preconditions.checkNotNull(distributionDesc);
+        DistributionInfo distributionInfo = distributionDesc.toDistributionInfo(columns);
+        if (distributionInfo.getBucketNum() == 0) {
+            int numBucket = CatalogUtils.calBucketNumAccordingToBackends();
+            distributionInfo.setBucketNum(numBucket);
+        }
+
+        // Refresh
+        MaterializedView.MvRefreshScheme mvRefreshScheme = new MaterializedView.MvRefreshScheme();
+        mvRefreshScheme.setType(MaterializedView.RefreshType.INCREMENTAL);
+
+        String mvName = stmt.getTableName().getTbl();
+        return new MaterializedView(mvId, dbId, mvName, columns, stmt.getKeysType(), partitionInfo,
+                distributionInfo, mvRefreshScheme);
+    }
+
+    public static void createIMT(CreateMaterializedViewStatement stmt, MaterializedView view) throws DdlException {
+        ExecPlan maintenancePlan = stmt.getMaintenancePlan();
+        OptExpression optExpr = maintenancePlan.getPhysicalPlan();
+        List<CreateTableStmt> createTables = IMTCreatorVisitor.createIMTForOperator(stmt, optExpr, view);
+
+        for (CreateTableStmt create : createTables) {
+            LOG.info("creating IMT {} for MV {}", create.getTableName(), view.getName());
+            try {
+                GlobalStateMgr.getCurrentState().createTable(create);
+            } catch (DdlException e) {
+                // TODO(murphy) cleanup created IMT, or make it atomic
```

Comment:
Where we can know: which IMTTables are created in an Incremental MV?  when we drop the IncrementalMV, the according StateTables should be dropped?

Reply:
All these information should be recorded in the `MVMaintenanceJob` and be persisted through editlog.